src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,41 @@
+package duke.task;
+
+import duke.format.DateFormat;
+import java.util.Date;
+
+
+/**
+ * Represents a deadline type of task that contains an additional due date.
+ */
+public class Deadline extends Task {
+    protected Date by;
+
+    /**
+     * Creates an instance of deadline
+     * @param name the name of deadline
+     * @param isComplete the completion status of deadline
+     * @param by the due date of deadline
+     */
+    public Deadline(String name, boolean isComplete, Date by) {
```

Comment:
maybe set isComplete false by default so no need to assign isComplete every time making deadline (same applies for other task related class)
ex) new Deadline("task", false, 2020-9-1) => new Deadline("task", 2020-9-1)